catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/forms/builder/XmlModelBuilder.java

```diff
@@ -109,87 +75,194 @@ public JAXBElement getResult() {
-    verifyResultNotYetRetrieved();
-    verifyTerminalNodeNotInProgress();
-    verifyLogicalNodeInProgress();
-    verifyLogicalNodeHasChildren();
-    JAXBElement result = logicOpCache.pop().apply(depth.pop());
-    if (!depth.isEmpty()) {
-      depth.peek().add(result);
-    } else {
-      rootNode = result;
+  public XmlModelBuilder or() {
+    beginBinaryLogicType(Mapper::or);
+    return this;
+  }
+
+  @Override
+  public XmlModelBuilder end() {
+    if (supplierInProgress != null) {
+      if (supplierInProgress.getParent() != null) {
+        endFunctionType();
+        return this;
+      }
+      endTerminalType();
+      return this;
     }
+    endBinaryLogicType();
     return this;
   }
 
-  // Note: Currently taking in JSON type symbol as the "operator"
   @Override
-  public XmlModelBuilder beginBinaryComparisonType(String operator) {
-    verifyResultNotYetRetrieved();
-    verifyTerminalNodeNotInProgress();
-    MultiNodeReducer comparisonMapping = TERMINAL_OPS.get(operator);
-    if (comparisonMapping == null) {
-      throw new IllegalArgumentException(
-          "Cannot find mapping for binary comparison operator: " + operator);
+  public XmlModelBuilder isEqualTo(boolean matchCase) {
+    beginTerminalType(Mapper::equalTo);
+    return this;
+  }
+
+  @Override
+  public XmlModelBuilder isNotEqualTo(boolean matchCase) {
+    beginTerminalType(Mapper::notEqualTo);
+    return this;
+  }
+
+  @Override
+  public XmlModelBuilder isGreaterThan(boolean matchCase) {
+    beginTerminalType(Mapper::greaterThan);
+    return this;
+  }
+
+  @Override
+  public XmlModelBuilder isGreaterThanOrEqualTo(boolean matchCase) {
+    beginTerminalType(Mapper::greaterThanOrEqualTo);
+    return this;
+  }
+
+  @Override
+  public XmlModelBuilder isLessThan(boolean matchCase) {
+    beginTerminalType(Mapper::lessThan);
+    return this;
+  }
+
+  @Override
+  public XmlModelBuilder isLessThanOrEqualTo(boolean matchCase) {
+    beginTerminalType(Mapper::lessThanOrEqualTo);
+    return this;
+  }
+
+  @Override
+  public XmlModelBuilder like(
+      boolean matchCase, String wildcard, String singleChar, String escape) {
+    if (matchCase) {
+      beginTerminalType(Mapper::likeMatchCase);
+      return this;
     }
-    supplierInProgress = new TerminalNodeSupplier(comparisonMapping);
+    beginTerminalType(Mapper::like);
     return this;
   }
 
   @Override
-  public FlatFilterBuilder beginPropertyIsLikeType(String operator, boolean matchCase) {
+  public XmlModelBuilder before() {
+    beginTerminalType(Mapper::before);
+    return this;
+  }
+
+  @Override
+  public XmlModelBuilder after() {
+    beginTerminalType(Mapper::after);
+    return this;
+  }
+
+  @Override
+  public XmlModelBuilder intersects() {
+    beginTerminalType(Mapper::intersects);
+    return this;
+  }
+
+  @Override
+  public XmlModelBuilder dwithin(double distance, String units) {
+    beginTerminalType(Mapper::dwithin);
+    return this;
```

Comment:
✏️ This is currently not a supported operation. Might revert this support for consistency and add it back in later.

Reply:
Removed. This code isn't even technically reachable anyway.